src/main/java/de/cronn/validation_files_diff/impl/MyRefactoringElementListenerProviderImpl.java

```diff
@@ -0,0 +1,89 @@
+package de.cronn.validation_files_diff.impl;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import com.intellij.psi.*;
+import com.intellij.refactoring.listeners.RefactoringElementAdapter;
+import com.intellij.refactoring.listeners.RefactoringElementListener;
+import com.intellij.refactoring.listeners.RefactoringElementListenerProvider;
+import com.intellij.refactoring.rename.RenameProcessor;
+import com.intellij.refactoring.rename.RenamePsiElementProcessor;
+import com.intellij.refactoring.rename.RenamePsiFileProcessor;
+import com.intellij.usageView.UsageInfo;
+import de.cronn.validation_files_diff.ValidationDiffProjectOptionsProvider;
+import de.cronn.validation_files_diff.helper.ModuleAnalyser;
+import de.cronn.validation_files_diff.helper.PsiTestNameUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class MyRefactoringElementListenerProviderImpl implements RefactoringElementListenerProvider {
+	@Override
+	public @Nullable RefactoringElementListener getListener(PsiElement element) {
+		if (!(element instanceof PsiMethod)) {
+			return null;
+		}
+
+
+		PsiMethod psiMethod = (PsiMethod) element;
+		PsiClass psiClass = psiMethod.getContainingClass();
+
+		String oldTestName = PsiTestNameUtils.getTestName(psiClass, psiMethod);
+		Path matchingContentRootForNextNonLeafModule = getMatchingContentRootForNextNonLeafModule(element);
+		VirtualFile moduleRoot = getLocalFileSystem().findFileByNioFile(matchingContentRootForNextNonLeafModule);
+
+		ValidationDiffProjectOptionsProvider instance = ValidationDiffProjectOptionsProvider.getInstance(element.getProject());
+		List<VirtualFile> virtualFiles = List.of(instance.getRelativeOutputDirPath(), instance.getRelativeValidationDirPath())
+				.stream()
+				.map(moduleRoot::findFileByRelativePath)
+				.filter(Objects::nonNull)
+				.map(VirtualFile::getChildren)
+				.flatMap(Arrays::stream)
+				.filter(v -> matches(v, oldTestName))
```

Comment:
Maybe keep a bit more semantic in the name, like `fileNameStartsWith(..)`.

Reply:
done, a3adb112ca046e1ba64cc32c1d6bdf9548d02391